hypothesis-python/src/hypothesis/_settings.py

```diff
@@ -523,47 +425,27 @@ class Setting(object):
 )
 
 
-settings._define_setting(
-    "max_shrinks",
-    default=not_set,
-    description="""
-Passing ``max_shrinks=0`` disables the shrinking phase (see the ``phases``
-setting), but any other value has no effect and uses a general heuristic.
-""",
-    deprecation_message="""
-The max_shrinks setting has been disabled, as internal heuristics are more
-useful for this purpose than a user setting.
-""",
-    deprecated_since="2018-04-14",
-)
-
-
 def _validate_timeout(n):
+    assert n is not not_set, "Validation is skipped if the value is not_set"
     if n is unlimited:
-        return -1
-    else:
+        note_deprecation(
+            "The timeout setting has been removed.  Please delete it from your "
+            "settings, and optionally configure the `deadline` setting to replace it.",
```

Comment:
Same comment as before that recommending deadline as a replacement for timeout is highly misleading.